hal/tests/TESTS/pin_names/pinvalidate.py

```diff
@@ -46,286 +79,392 @@ def find_target(target_name = ''):
+            ):
                 error_count += 1
-        table.append([target, result, error_count])
-    
-    return tabulate(table, headers=['Platform name', 'Result', 'Error count'], tablefmt='grid')
+
+        table.append(
+            [target_name, "FAILED" if error_count else "PASSED", error_count]
+        )
+
+    return tabulate(
+        table,
+        headers=["Platform name", "Result", "Error count"],
+        tablefmt="grid",
+    )
+
 
 def print_suite_summary(report):
     targets = []
     for case in report:
-        targets.append(case['platform_name'])
+        targets.append(case["platform_name"])
     targets = list(dict.fromkeys(targets))
-    
+
     table = []
     for target in targets:
         suites = []
         for case in report:
-            if case['platform_name'] != target:
+            if case["platform_name"] != target:
                 continue
-            suites.append(case['suite_name'])
+            suites.append(case["suite_name"])
         suites = list(dict.fromkeys(suites))
 
         for suite in suites:
-            result = 'PASSED'
+            result = "PASSED"
             error_count = 0
             for case in report:
-                if case['platform_name'] != target:
+                if case["platform_name"] != target:
                     continue
-                if case['suite_name'] != suite:
+                if case["suite_name"] != suite:
                     continue
-                if case['result'] == 'FAILED':
-                    result = 'FAILED'
+                if case["result"] == "FAILED":
+                    result = "FAILED"
                     error_count += 1
             table.append([target, suite, result, error_count])
-    
-    return tabulate(table, headers=['Platform name', 'Test suite', 'Result', 'Error count'], tablefmt='grid')
 
-def print_report(report, print_error_detail, tablefmt='grid'):
+    return tabulate(
+        table,
+        headers=["Platform name", "Test suite", "Result", "Error count"],
+        tablefmt="grid",
+    )
+
+
+def print_report(report, print_error_detail, tablefmt="grid"):
     table = []
     for case in report:
         errors_str = ""
-        for error in case['errors']:
-            errors_str += '\n'
-            if error['key']:
-                errors_str += error['key']
-            if error['val']:
-                errors_str += " = " + error['val']
-            if error['message']:
-                errors_str += " <-- " + error['message']
-        
-        if errors_str == '':
-            errors_str = 'None'
+        for error in case["errors"]:
+            errors_str += "\n"
+            if error["key"]:
+                errors_str += error["key"]
+            if error["val"]:
+                errors_str += " = " + error["val"]
+            if error["message"]:
+                errors_str += " <-- " + error["message"]
+
+        if errors_str == "":
+            errors_str = "None"
```

Comment:
If there are errors in this, would there be a case where there is not a `key` or `message`? I saw earlier that there might not be a `value`. If not, you could potentially drop some of the if statements.

Also, the more I think of it, the more I am confusing myself with a dictionary key named `"key"`. That might be pedantic, but it is throwing me off a little.

Reply:
Currently all test cases output a key and message but in the future there may be test cases that don't.